frappe/model/delete_doc.py

```diff
@@ -108,16 +108,16 @@ def delete_doc(
 				update_flags(doc, flags, ignore_permissions)
 				check_permission_and_not_submitted(doc)
 
-				if not ignore_on_trash:
-					doc.run_method("on_trash")
-					doc.flags.in_delete = True
-					doc.run_method("on_change")
-
 				# check if links exist
 				if not force:
 					check_if_doc_is_linked(doc)
 					check_if_doc_is_dynamically_linked(doc)
 
+				if not ignore_on_trash:
+					doc.run_method("on_trash")
+					doc.flags.in_delete = True
+					doc.run_method("on_change")
+
```

Comment:
This is a big breaking change. `on_trash` in various places is used to clean up or update links manually.